credentials/tls_ext_test.go

```diff
@@ -79,43 +79,76 @@ func (s) TestTLS_MinVersion12(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
 	defer cancel()
 
-	// Create server creds without a minimum version.
-	serverCreds := credentials.NewTLS(&tls.Config{
+	serverTLS := &tls.Config{
 		// MinVersion should be set to 1.2 by gRPC by default.
 		Certificates: []tls.Certificate{serverCert},
-	})
-	ss := stubserver.StubServer{
-		EmptyCallF: func(context.Context, *testpb.Empty) (*testpb.Empty, error) {
-			return &testpb.Empty{}, nil
+	}
+	noOpGetCfgForClient := serverTLS.Clone()
+	noOpGetCfgForClient.GetConfigForClient = func(*tls.ClientHelloInfo) (*tls.Config, error) {
+		return nil, nil
+	}
+
+	testCases := []struct {
+		name      string
+		serverTLS *tls.Config
+	}{
+		{
+			name:      "base_case",
+			serverTLS: serverTLS,
+		},
+		{
+			name:      "fallback_to_base",
+			serverTLS: noOpGetCfgForClient,
```

Comment:
Optional:

```golang
	serverTLS: func() *tls.Config {
		cfg := serverTLS.Clone()
		cfg.GetConfigForClient = func(*tls.ClientHelloInfo) (*tls.Config, error) { return nil, nil }
		return cfg
	}()
```

This will get rid of the `noOpGetCfgForClient` var which is only used by this subtest.

Reply:
Nice idea, changed as suggested.